### PR TITLE
Keycloak validate token

### PR DESCRIPTION
**Pull request by sergiofilhowz**
KeycloakFilter now handles every request by validating if the keycloak token is active

### CODE REVIEW

**Comment by sergiofilhowz** [comment]
I'm also commiting the jar and war dependencies separated. To be able to import it as jar and use the internal API.

File: core/src/main/java/org/entando/entando/keycloak/filter/KeycloakFilter.java [hunk 1]
```diff
@@ -67,12 +75,61 @@ public void doFilter(final ServletRequest servletRequest,
...
+    }
+
+    private boolean isAccessTokenValid(final String accessToken) {
+        final ResponseEntity<AccessToken> tokenResponse = oidcService.validateToken(accessToken);
+        return HttpStatus.OK.equals(tokenResponse.getStatusCode())
+                && tokenResponse.getBody() != null
+                && tokenResponse.getBody().isActive();
+    }
+
+    private boolean refreshToken(final HttpServletRequest request) {
```
Comment (joewhite101) [inline core/src/main/java/org/entando/entando/keycloak/filter/KeycloakFilter.java]: Does this have unit test coverage

File: core/src/main/java/org/entando/entando/keycloak/filter/KeycloakFilter.java [hunk 2]
```diff
@@ -67,12 +75,61 @@ public void doFilter(final ServletRequest servletRequest,
...
         if ("/do/login".equals(request.getServletPath()) || "/do/login.action".equals(request.getServletPath())) {
             doLogin(request, response, chain);
         } else if ("/do/logout.action".equals(request.getServletPath())) {
             doLogout(request, response);
+        } else {
+            chain.doFilter(request, response);
+        }
+    }
+
+    private boolean isAccessTokenValid(final String accessToken) {
```
Comment (joewhite101) [inline core/src/main/java/org/entando/entando/keycloak/filter/KeycloakFilter.java]: Does this have test coverage?

File: core/src/main/java/org/entando/entando/keycloak/services/oidc/OpenIDConnectService.java [hunk 3]
```diff
@@ -131,6 +131,17 @@ public String getRedirectUrl(final String redirectUri, final String state) throw
         return new HttpEntity<>(body, headers);
     }
 
+    private HttpEntity<MultiValueMap<String, String>> createRefreshTokenRequest(final String refreshToken) {
```
Comment (joewhite101) [inline core/src/main/java/org/entando/entando/keycloak/services/oidc/OpenIDConnectService.java]: Confirm test coverage

Comment (sergiofilhowz) [reply core/src/main/java/org/entando/entando/keycloak/services/oidc/OpenIDConnectService.java]: added a refresh token test

**Pull request merged**